ssl-config/src/main/java/com/palantir/ssl/SslConfiguration.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2015 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.ssl;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.google.common.base.Optional;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import org.immutables.value.Value;
+import org.immutables.value.Value.Style.ImplementationVisibility;
+
+@JsonDeserialize(as = ImmutableSslConfiguration.class)
+@JsonSerialize(as = ImmutableSslConfiguration.class)
+@Value.Immutable
+@Value.Style(visibility = ImplementationVisibility.PACKAGE)
+public abstract class SslConfiguration {
```

Comment:
My initial thoughts are to prefer separate JKS files because merging them
is not easy.  However if we expect deployment services to lay down these
files in the future it kind of doesn't matter when you control both.

On Mon, Feb 8, 2016 at 1:42 PM, Mark Elliot <notifications@github.com>
wrote:

> In ssl-config/src/main/java/com/palantir/ssl/SslConfiguration.java
> <https://github.com/palantir/http-remoting/pull/35#discussion_r52225002>:
>
> > +
> > +import static com.google.common.base.Preconditions.checkArgument;
> > +
> > +import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
> > +import com.fasterxml.jackson.databind.annotation.JsonSerialize;
> > +import com.google.common.base.Optional;
> > +import java.nio.file.Path;
> > +import java.nio.file.Paths;
> > +import org.immutables.value.Value;
> > +import org.immutables.value.Value.Style.ImplementationVisibility;
> > +
> > +@JsonDeserialize(as = ImmutableSslConfiguration.class)
> > +@JsonSerialize(as = ImmutableSslConfiguration.class)
> > +@Value.Immutable
> > +@Value.Style(visibility = ImplementationVisibility.PACKAGE)
> > +public abstract class SslConfiguration {
>
> I wonder if we should consider having just a single JKS file rather than 2
> JKS files (or two references to the same file).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/palantir/http-remoting/pull/35/files#r52225002>.
>